core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -338,6 +327,7 @@ import scala.util.control.NonFatal
       commitRefreshing.committed(offsets)
 
     case Stop =>
+      maybeCommit()
```

Comment:
That's a good question.  If commits are in process then the `KafkaConsumerActor` will start using the `stopping` behaviour, but that behavior will only do up to 1 more poll (and possibly handle some or all remaining commit callbacks) and then stop the actor.  I think I assumed the polling would continue until all commit callbacks have returned, but since that's not the case I agree that doing a `commitSync` would make more sense.

Do you know what the purpose of allowing 1 final poll when in `stopping`?

Reply:
It is just that: collect replies to any outstanding commits.
And that solution is actually better now that I play around with it, as it doesn't block.

Reply:
Yes, I misread part of the condition that checks if there are still comits in progress.  So it _will_ poll until all callbacks have returned.

```
if (stopInProgress && commitsInProgress == 0) {
```

So to be clear, you're fine with this implementaiton as-is?